### PR TITLE
Automatically create and publish Docker image

### PR DESCRIPTION
**Pull request by imikushin**
Also, remove hardcoded Rust toolchain (use `rust-toolchain.toml`) in CI.

### CODE REVIEW

File: .github/workflows/build-image.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
+name: Create and publish a Docker image
+
+on:
+  push:
+    branches: ['imikushin/automate-image-build'] # TODO change to main
```
Comment (mimoo) [inline .github/workflows/build-image.yml:5]: change to main I guess :D

Comment (mimoo) [reply .github/workflows/build-image.yml:5]: altho should we maybe not do this on every push, but rather for releases specifically?

https://docs.github.com/en/actions/publishing-packages/publishing-docker-images has this example:

```yml
on:
  release:
    types: [published]
```

**Review by mimoo — APPROVED** [review]
Nice!

File: .github/workflows/build-image.yml (hunk 1 above)
Comment (imikushin) [reply .github/workflows/build-image.yml:5]: > change to main I guess :D

Will do: I'm still experimenting 🙃

Comment (imikushin) [reply .github/workflows/build-image.yml]: Updated: only pushing to `release` will trigger building and publishing an image.

**Pull request merged**